crates/credentials/src/vc.rs

```diff
@@ -127,25 +195,163 @@ impl VerifiableCredential {
 
-        // TODO Implement semantic VC verification rules https://github.com/TBD54566975/web5-rs/issues/151
+        // registered claims checks
+        if registered_claims.issuer.is_none() {
+            return Err(CredentialError::MissingIss);
+        }
+        if registered_claims.subject.is_none() {
+            return Err(CredentialError::MissingSub);
+        }
+        if registered_claims.not_before.is_none() {
+            return Err(CredentialError::MissingNbf);
+        }
+        if registered_claims.jti.is_none() {
+            return Err(CredentialError::MissingJti);
+        }
 
-        Ok(jwt_decoded.claims.vc)
+        // TODO: Change this to None after this issue is resolved - https://github.com/TBD54566975/web5-rs/issues/202
+        // check claims match expected values
+        if vc.id == String::default() {
+            vc.id = registered_claims.jti.clone().unwrap();
+        } else if vc.id != registered_claims.jti.unwrap() {
+            return Err(CredentialError::IdMismatch);
+        }
+
+        if vc.credential_subject.id == String::default() {
+            vc.credential_subject.id = registered_claims.subject.clone().unwrap();
+        } else if vc.credential_subject.id != registered_claims.subject.unwrap() {
+            return Err(CredentialError::SubjectMismatch);
+        }
+
+        if vc.issuance_date == i64::default() {
+            vc.issuance_date = registered_claims.not_before.unwrap();
+        } else if vc.issuance_date != registered_claims.not_before.unwrap() {
+            return Err(CredentialError::IssuanceDateMismatch);
+        }
+
+        if vc.expiration_date.unwrap() == i64::default() {
+            vc.expiration_date = Some(registered_claims.expiration.unwrap())
+        } else if vc.expiration_date != Some(registered_claims.expiration.unwrap()) {
+            return Err(CredentialError::ExpirationDateMismatch);
+        }
+
+        // issuer check
+        let iss = registered_claims.issuer.as_ref().unwrap();
+        match &mut vc.issuer {
+            Issuer::String(id) => {
+                if id.is_empty() {
+                    id.clone_from(iss);
+                } else if id != iss {
+                    return Err(CredentialError::IssuerMismatch);
+                }
+            }
+            Issuer::Object(named) => {
+                if &named.id != iss {
+                    return Err(CredentialError::IssuerMismatch);
+                }
+            }
+        }
+
+        // optional expiration date check
+        if let (Some(exp), Some(vc_exp)) = (registered_claims.expiration, vc.expiration_date) {
+            if vc_exp == i64::default() {
+                vc.expiration_date = Option::from(exp);
+            } else if exp != vc_exp {
+                return Err(CredentialError::ExpirationDateMismatch);
+            }
+
+            let now = SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_secs() as i64;
```

Comment:
again I'd advise we stay away from calling `unwrap` in non-test code for the sake of error handling. also we should avoid using `as` casting as there's a baked in assumption there, instead we can use `i64::try_from()`

```suggestion
            let now = i64::try_from(SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs())?;
```

And then you'll have to add new error enums (and add the relevant `use` statements)

```rust
#[derive(thiserror::Error, Debug)]
pub enum CredentialError {
    #[error(transparent)]
    SystemTimeError(#[from] SystemTimeError),
    #[error(transparent)]
    TryFromIntError(#[from] TryFromIntError),
    // ...
```

Reply:
Lol lets use chrono :

`let now = Utc::now().timestamp();`

Reply:
nice!